core/src/main/java/org/infinispan/functional/impl/Params.java

```diff
@@ -108,4 +115,50 @@ public static Params from(Param<?>... ps) {
       }
    }
 
+   public static class Externalizer implements AdvancedExternalizer<Params> {
+      @Override
+      public Set<Class<? extends Params>> getTypeClasses() {
+         return Collections.singleton(Params.class);
+      }
+
+      @Override
+      public Integer getId() {
+         return Ids.PARAMS;
+      }
+
+      @Override
+      public void writeObject(ObjectOutput output, Params object) throws IOException {
+         for (Param param : object.params) {
+            switch (param.serializeMethod()) {
```

Comment:
@galderz So you think that Params should already know about all possible Param implementations (it does, through defaults), and serialize them one-by-one? That's what I've tried to avoid, by delegating the information about the params to themselves.
I want to spare the bytes and lookups in marshaller when the params always have the same order (according to their ids) and some of them don't make sense to be serialized at all (such as the future mode which is purely local thing).